examples/vae/main.py

```diff
@@ -142,30 +139,37 @@ def eval(model, eval_ds, z):
 
 
 def main(argv):
-  key = random.PRNGKey(0)
+  del argv
+  rng = random.PRNGKey(0)
+  rng, key = random.split(rng)
+
   train_ds = tfds.load('mnist', split=tfds.Split.TRAIN)
   train_ds = train_ds.cache().shuffle(1000).batch(FLAGS.batch_size)
   test_ds = tfds.as_numpy(tfds.load('mnist', split=tfds.Split.TEST, batch_size=-1))
 
-  _, params = VAE.init_by_shape(key, [((1, 784), jnp.float32)])
+  _, params = VAE.init_by_shape(
+      key, [((1, 784), jnp.float32)], z_rng=random.PRNGKey(0))
```

Comment:
It doesn't make a difference for this specific model, but I think it'd be cleaner to specify the actual batch dimension during model init (instead of dim 1 as it is now).

Reply:
done